### PR TITLE
Remove allowlist for local dependency replacement

### PR DESCRIPTION
**Pull request by lkolacek**
Remove allowlist for local dependency replacement and add integration-test

It is been decided in CLOUDBLD-11314 that we will not port allowlists from Cachito to Cachi2 to moderate  every dependency replacement for gomod, pip, npm/yarn etc.

CLOUDBLD-12781, CLOUDBLD-12631

# Maintainers will complete the following section

- [x] Commit messages are descriptive enough
- [x] Code coverage from testing does not decrease and new code is covered
- n/a New code has type annotations
- n/a Docs updated (if applicable)
- n/a Docs links in the code are still valid (if docs were updated)

### CODE REVIEW

File: cachi2/core/package_managers/gomod.py [hunk 1]
```diff
@@ -771,7 +746,7 @@ def _vet_local_deps(dependencies: List[dict], module_name: str, allowed_patterns
             )
```
Comment (chmeliik) [inline cachi2/core/package_managers/gomod.py:746]: IMO we can remove the `Module <module> wants to replace <package> with a local dependency: <path>`  log message as well, it's just noise at this point

File: tests/integration/test_package_managers.py [hunk 2]
```diff
@@ -90,6 +90,19 @@
             ),
             id="gomod_vendor_check_correct_vendor",
         ),
+        # Test case checks if package can be replaced with local dependency
+        pytest.param(
+            utils.TestParameters(
+                repo="https://github.com/cachito-testing/cachito-gomod-local-deps.git",
+                ref="b2e465b91a6a272540c77d4dde1e317773ed700b",
+                packages=({"path": ".", "type": "gomod"},),
+                flags=["--gomod-vendor-check"],
```
Comment (chmeliik) [inline tests/integration/test_package_managers.py:99]: is `--gomod-vendor-check` an intended part of the test?

**Comment by brunoapimentel** [comment]
I think we could drop a quick note on the commit message as to why we're removing the allow list.

File: tests/integration/test_package_managers.py (hunk 2 above)
Comment (lkolacek) [reply tests/integration/test_package_managers.py]: Nice catch! This was an accidental copy ..

File: cachi2/core/package_managers/gomod.py (hunk 1 above)
Comment (lkolacek) [reply cachi2/core/package_managers/gomod.py]: log message removed

**Comment by lkolacek** [comment]
Thanks @brunoapimentel @chmeliik. I addressed your comments.

File: cachi2/core/package_managers/gomod.py [hunk 3]
```diff
@@ -733,45 +728,16 @@ def _load_list_deps(list_deps_output: str) -> Dict[str, dict]:
...
-    return allowed_deps or []
-
-
-def _vet_local_deps(dependencies: List[dict], module_name: str, allowed_patterns: List[str]):
-    """
-    Fail if any dependency is replaced by a local path unless the module is allowlisted.
-
-    Also fail if the module is allowlisted but the path is absolute or outside repository.
-    """
+def _vet_local_deps(dependencies: List[dict], module_name: str):
```
Comment (chmeliik) [inline cachi2/core/package_managers/gomod.py:731]: If I'm not mistaken, the `module_name` argument is unused now

**Review by chmeliik — COMMENTED** [review]
LGTM, minor nitpick about an unused argument

**Review by ejegrova — APPROVED** [review]
LGTM

**Review by chmeliik — APPROVED** [review]

**Pull request merged**